src/tests/validity/variable_constant_condition.ab

```diff
@@ -0,0 +1,22 @@
+// Output
+// Outer: 42
+// If: 12
+// Else: 24
+
+main {
+    const x = 42
+    echo "Outer: {x}"
+
+    if x > 12 {
+        const x = 12
+        echo "If: {x}"
+    }
+
```

Comment:
Consider testing the outer value again, to confirm that the constant declaration does not overwrite that value in the same function scope.